tests/library/global-fetch.spec.ts

```diff
@@ -96,6 +96,47 @@ it('should propagate extra http headers with redirects', async ({ playwright, se
   expect(req3.headers['my-secret']).toBe('Value');
 });
 
+it('should throw an error when max redirects are exceeded', async ({ playwright, server }) => {
+  server.setRedirect('/a/redirect1', '/b/c/redirect2');
+  server.setRedirect('/b/c/redirect2', '/b/c/redirect3');
+  server.setRedirect('/b/c/redirect3', '/b/c/redirect4');
+  server.setRedirect('/b/c/redirect4', '/simple.json');
+
+  const request = await playwright.request.newContext();
+  for (const follow of [1, 2]) await expect.soft(async () => request.get(`${server.PREFIX}/a/redirect1`, { follow: follow })).rejects.toThrow('Max redirect count exceeded');
```

Comment:
All the asserts should be hard, we don't need to continue if any operation fails.